### PR TITLE
fix(TDP-4618): timeout issues when opening a preparation if no cache is available

### PR DESCRIPTION
**Pull request by fabballe**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-4618

**Please check if the PR fulfills these requirements**
- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [ ] The new code does not introduce new technical issues (sonar / eslint)
- [ ] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/6/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/6/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/16/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/16/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/21/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/21/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/25/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/25/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/26/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/26/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/30/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/30/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/33/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/33/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/34/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/34/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/35/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/35/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/36/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/36/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/37/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/37/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/39/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/39/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/43/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/43/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/45/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/45/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/62/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/62/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/63/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/63/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/68/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/68/)

File: dataprep-webapp/src/app/services/rest/rest-queued-message-interceptor-factory.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+};
+const ALLOWED_METHODS = [METHODS.POST, METHODS.GET, METHODS.HEAD];
+const NOOP = () => {};
+
+/**
+ * @ngdoc service
+ * @name data-prep.services.rest.service:RestQueuedMessageHandler
+ * @description Queued message interceptor
+ */
+export default function RestQueuedMessageHandler($q, $injector, $timeout, RestURLs) {
```
Comment (charlesnguyen) [inline dataprep-webapp/src/app/services/rest/rest-queued-message-interceptor-factory.js]: Could you add more unit tests to cover at least nominal use cases, please?

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/104/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/104/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/107/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/107/)

File: dataprep-webapp/src/app/services/rest/rest-queued-message-interceptor-factory.js (hunk 1 above)
Comment (ncomont) [reply dataprep-webapp/src/app/services/rest/rest-queued-message-interceptor-factory.js]: :white_check_mark:

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/118/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/118/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/120/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/120/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/123/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/123/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/129/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/129/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/130/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/130/)

**Pull request merged**